### ISSUE TITLE
[FEATURE REQUEST] Multi Threaded YT Downloads

### ISSUE THREAD

**Issue by iWisp360** [namidaco/namida#226]
**Is your feature request related to a problem? Please describe.**

I have slow and limited internet, one download thread is not enough.

**Describe the solution you'd like**
Multithreaded YT downloads by default to improve download speeds

**Apps**
- Literally any good download manager out there
- Newpipe

**Comment by MSOB7YY** [comment]
this will introduce lotta issues, the current implementation literally depends on the file size (bytes) in order to resume downloads. multi threads will write to multiple byte offset, making the current implementation of no use.
another way is to split downloads to multiple files but that still requires re-write and of course the bugs are awaiting.

THE THING IS multi threading is not always the solution. it helps only when the server is throttling your connection so u open multiple connections. having slow internet will not really help here

---

TLDR; might consider it but only on the long run, when most things are already stable enough 
---
